File: interfaces/org/opentravelmate/widget/mainmenu/v1/MenuItem.js
/**
 * Main menu item.
 *
 * @author marc.plouhinec@gmail.com (Marc Plouhinec)
 */



/**
 * Create a menu item.
 *
 * @param {!string} title
 *   Menu item title.
 * @param {?string} tooltip
 *   Text displayed when the user hovers the mouse pointer on top of this
 *   menu item.
 * @param {!string} iconUrl
 *   URL to the menu item icon.
 * @constructor
 */
function MenuItem(title, tooltip, iconUrl) {
  'use strict';

  /**
   * Menu item title.
   *
   * @type {!string}
   */
  this.title = title;

  /**
   * Menu item tooltip.
   *
   * @type {?string}
   */
  this.tooltip = tooltip;

  /**
   * Menu item icon URL.
   *
   * @type {!string}
   */
  this.iconUrl = iconUrl;

  /**
   * Registered click listeners.
   *
   * @private
   * @type {Array.<function>}
   */
  this.clickListeners_ = [];
}


/**
 * Add a click listener.
 *
 * @param {!function(this: MenuItem)} clickListener
 *   Function called when the user clicks on the menu item.
 *   Note: the value of 'this' in the function is the current MenuItem.
 */
MenuItem.prototype.onClick = function(clickListener) {
  'use strict';

  this.clickListeners_.push(clickListener);
};


/**
 * Fire a click event to the registered listeners.
 */
MenuItem.prototype.fireOnClickEvent = function() {
  'use strict';

  var /** @type {number} */ i;
  for (i = 0; i < this.clickListeners_.length; i += 1) {
    this.clickListeners_[i].apply(this);
  }
};


/** Export the class as a module */
module.exports = MenuItem;

File: android/app/src/org/opentravelmate/widget/menu/MenuItem.java
package org.opentravelmate.widget.menu;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Define a menu item.
 *
 * @author marc.plouhinec@gmail.com (Marc Plouhinec)
 */
public class MenuItem {
	
	/**
	 * Item ID.
	 */
	public final int id;
	
	/**
	 * Item title.
	 */
	public final String title;
	
	/**
	 * Item tool tip (displayed when the user put the mouse over the item).
	 */
	public final String tooltip;
	
	/**
	 * Item icon URL.
	 */
	public final String iconUrl;
	
	/**
	 * Create a menu item.
	 * 
	 * @param id
	 * @param title
	 * @param tooltip
	 * @param iconUrl
	 */
	public MenuItem(int id, String title, String tooltip, String iconUrl) {
		this.id = id;
		this.title = title;
		this.tooltip = tooltip;
		this.iconUrl = iconUrl;
	}
	
	/**
	 * Build a MenuItem from a JSON-serialized representation.
	 * 
	 * @param jsonMenuItem
	 * @return MenuItem
	 * @throws JSONException
	 */
	public static MenuItem fromJsonMenuItem(JSONObject jsonMenuItem) throws JSONException {
		int id = jsonMenuItem.getInt("id");
		String title = jsonMenuItem.getString("title");
		String tooltip = jsonMenuItem.getString("tooltip");
		String iconUrl = jsonMenuItem.getString("iconUrl");
		return new MenuItem(id, title, tooltip, iconUrl);
	}

	@Override
	public String toString() {
		return "MenuItem [id=" + id + ", title=" + title + ", tooltip="
				+ tooltip + ", iconUrl=" + iconUrl + "]";
	}
}
